src/App.tsx

```diff
@@ -36,60 +36,107 @@ export function getReorderedGoods(
 
   // Sort and reverse goods if needed
   // eslint-disable-next-line no-console
-  console.log(sortType, isReversed);
+  if (sortType !== SortType.NONE) {
+    visibleGoods.sort((firstGood, secondGood) => {
+      if (sortType === SortType.ALPHABET) {
+        return firstGood.localeCompare(secondGood);
+      }
+
+      if (sortType === SortType.LENGTH) {
+        return firstGood.length - secondGood.length;
+      }
+
+      return 0; // this is to make linter shut the hell up
+    });
+  }
+
+  if (isReversed) {
+    visibleGoods.reverse();
+  }
 
   return visibleGoods;
 }
 
-// DON'T save goods to the state
-// type State = {
-//   isReversed: boolean,
-//   sortType: SortType,
-// };
-
-export const App: React.FC = () => {
-  return (
-    <div className="section content">
-      <div className="buttons">
-        <button
-          type="button"
-          className="button is-info is-light"
-        >
-          Sort alphabetically
-        </button>
-
-        <button
-          type="button"
-          className="button is-success is-light"
-        >
-          Sort by length
-        </button>
-
-        <button
-          type="button"
-          className="button is-warning is-light"
-        >
-          Reverse
-        </button>
-
-        <button
-          type="button"
-          className="button is-danger is-light"
-        >
-          Reset
-        </button>
-      </div>
+type State = {
+  isReversed: boolean,
+  sortType: SortType,
+};
```

Comment:
you don't need to use State
```suggestion
```

Reply:
This isn't the version with the hooks, why shouldn't I use state?

Reply:
Nevermind, I saw the comment below, I feel dumb now